### PR TITLE
[WIP] upload binary to bintray for every PR

### PR DESCRIPTION
**Pull request by kadel**
For every PR build binary is uploaded to bintray to make it easy for anyone to test PR without a need to rebuild manually.

Followup to this will be creating a script that will automatically post a comment to PR with a link to where binary can be downloaded.

### CODE REVIEW

**Comment by cdrage** [comment]
I don't see any glaring issues with this, this LGTM.

**Comment by anmolbabu** [comment]
@kadel, Not sure if I am missing something... but, aren't we uploading the packages to a separate dir(for pr builds) on bintray because I don't see any dir by name of form https://travis-ci.org/redhat-developer/odo/jobs/440625179#L654 on https://dl.bintray.com/odo/ (Am I looking at the correct url?) in any of its internal dirs..

**Comment by surajnarwade** [comment]
this looks good to me :)
@kadel , can you we do something like this: `odo bot will comment on respective PR with generated binary link, so that reviewer can click on binary and pull it and test it`

**Comment by surajnarwade** [comment]
@kadel , oh I saw description of PR now :smiley_cat:

**Comment by kadel** [comment]
I had to redo this PR in a different way. 
Travis doesn't run deploy stage when PR is built, and it can't be forced.
I had to do it via script.

**Comment by kadel** [comment]
> @kadel, Not sure if I am missing something... but, aren't we uploading the packages to a separate dir(for pr builds) on bintray because I don't see any dir by name of form https://travis-ci.org/redhat-developer/odo/jobs/440625179#L654 on https://dl.bintray.com/odo/ (Am I looking at the correct url?) in any of its internal dirs..

It didn't work before :-( Deploy is stage is never run for PRs :-(
My last update is fixing that.

Now you should be able to find binaries at `https://dl.bintray.com/odo/odo/pr<number>`
for this PR - `https://dl.bintray.com/odo/odo/pr833`

File: .travis.yml [hunk 1]
```diff
@@ -79,9 +79,10 @@ jobs:
         - make prepare-release
         - ./scripts/generate-bintray-json.sh
         - make packages
-      script: skip
+      script: 
+        - ./scripts/upload-pr-binaries/sh
```
Comment (anmolbabu) [inline .travis.yml]: How does this work?
should it be ./scripts/upload-pr-binaries.sh instead ?

**Comment by kadel** [comment]
IT looks like this approach is not going to work, because of the security restrictions on travis-ci side - https://docs.travis-ci.com/user/pull-requests#pull-requests-and-security-restrictions

I'll have to look for another solutions that doesn't required storing any secrets

**Comment by kadel** [comment]
closed in favor of #898

**Pull request closed without merging**